### PR TITLE
input-date: fix tooltip jitter on click

### PR DESCRIPTION
**Pull request by margaree**
Fixes https://github.com/BrightspaceUI/core/issues/882

When `input-text` is clicked, it triggers the tooltip to be shown because of it being focused on. In order for the click not to show the tooltip, I added a timeout in the focus event listener for `input-text` where it waits to see if the `mouseup` event happens which would then set `_inputTextFocusMouseup` to true and in that case the tooltip should stay hidden. There doesn't seem to be a way to easily detect if the mouse vs keyboard caused the focus just within the focus listener, which would be more ideal.

### CODE REVIEW

File: components/inputs/test/input-date-time.visual-diff.js [hunk 1]
```diff
@@ -46,8 +46,10 @@ describe('d2l-input-date-time', () => {
 
 	it('basic-focus', async function() {
 		await page.$eval('#basic', (elem) => elem.focus());
-		const rect = await visualDiff.getRect(page, '#basic');
-		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+		setTimeout(async() => {
+			const rect = await visualDiff.getRect(page, '#basic');
+			await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+		}, 160);
```
Comment (dlockhart) [inline components/inputs/test/input-date-time.visual-diff.js:52]: Not sure if using the sinon fake timers here to quickly advance the time could make these run faster, or if it'll just make them not work at all. 🤷

File: components/inputs/input-date.js [hunk 2]
```diff
@@ -349,8 +353,19 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
...
+		this._inputTextFocusShowTooltip = true;
+	}
+
 	_handleInputTextFocus() {
 		this._formattedValue = this._shownValue ? formatISODateInUserCalDescriptor(this._shownValue) : '';
+
+		// hide tooltip when focus, wait to see if click happened, then show
+		this._inputTextFocusShowTooltip = false;
+		setTimeout(() => {
+			if (!this._inputTextFocusMouseup) this._inputTextFocusShowTooltip = true;
```
Comment (dlockhart) [inline components/inputs/input-date.js:367]: Crazy that there isn't a better way to detect the source (mouse vs. keyboard) of a focus event but you're right... I couldn't find another way either.

**Review by dlockhart — APPROVED** [review]
Change looks good, but the visual diffs look a little off. I think for some of them, we're now expecting the tooltip balloon triangle thing to go away, but there are more changes than just that.

File: components/inputs/test/input-date-time.visual-diff.js (hunk 1 above)
Comment (margaree) [reply components/inputs/test/input-date-time.visual-diff.js:52]: That's a good idea - I'll give it a go

**Review by dlockhart — APPROVED** [review]

**Pull request merged**